src/promptflow/promptflow/_sdk/entities/_utils.py

```diff
@@ -0,0 +1,41 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+"""
+Previously we have put some shared logic for flow in ProtectedFlow class. However, in practice,
+it's hard to guarantee that we always use this class to create flow; on the other hand, the more
+
+"""
+from os import PathLike
+from pathlib import Path
+from typing import Tuple, Union
+
+from promptflow._sdk._constants import DAG_FILE_NAME
+
+
+def get_flow_definition(
```

Comment:
maybe rename this to something like resolve_flow_path. 

get_flow_definition sounds like load yaml and return some definition class

Reply:
well, there is already a function named `resolve_flow_path` hh. let me try some other name